Bazel 9 regression

### Description of the bug:


Bazel 9 no longer supports "file://" scheme in http_archive:

```
http_archive(
    name = "large-movie-review-dataset",
    build_file = "external/nlp/large-movie-review-dataset.BUILD",
    sha256 = "c40f74a18d3b61f90feba1e17730e0d38e8b97c05fde7008942e91923d1658fe",
    url = "file:///abc/nlp/sentiment/en/aclImdb_v1.tar.gz",
)
```

```
Caused by: net.starlark.java.eval.Starlark$UncheckedEvalException: NullPointerException thrown during Starlark evaluation (repository @@+http_archive+large-movie-review-dataset)
        at <starlark>.download_and_extract(<builtin>:0)
        at <starlark>._http_archive_impl(/xyz/output/e039a0d51aa3648f6061154c43aa4253/external/bazel_tools/tools/build_defs/repo/http.bzl:200)
Caused by: java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "java.net.URI.getHost()" is null
        at com.google.devtools.build.lib.bazel.repository.downloader.ProgressInputStream.reportProgress(ProgressInputStream.java:139)
        at com.google.devtools.build.lib.bazel.repository.downloader.ProgressInputStream.read(ProgressInputStream.java:117)
        at com.google.devtools.build.lib.bazel.repository.downloader.HashInputStream.read(HashInputStream.java:62)
        at java.base/java.io.SequenceInputStream.read(Unknown Source)
        at java.base/java.io.FilterInputStream.read(Unknown Source)
        at java.base/java.io.InputStream.transferTo(Unknown Source)
        at com.google.devtools.build.lib.bazel.repository.downloader.HttpDownloader.download(HttpDownloader.java:101)
        at com.google.devtools.build.lib.bazel.repository.downloader.DelegatingDownloader.download(DelegatingDownloader.java:64)
        at com.google.devtools.build.lib.bazel.repository.downloader.DownloadManager.downloadInExecutor(DownloadManager.java:333)
        at com.google.devtools.build.lib.bazel.repository.downloader.DownloadManager.lambda$startDownload$0(DownloadManager.java:143)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPerTaskExecutor$ThreadBoundFuture.run(Unknown Source)
        at java.base/java.lang.VirtualThread.run(Unknown Source)
        Suppressed: com.google.devtools.build.lib.bazel.repository.downloader.ContentLengthMismatchException: Bytes read 19218432 but wanted 84125825
                at com.google.devtools.build.lib.bazel.repository.downloader.CheckContentLengthInputStream.checkContentLength(CheckContentLengthInputStream.java:71)
                at com.google.devtools.build.lib.bazel.repository.downloader.CheckContentLengthInputStream.close(CheckContentLengthInputStream.java:66)
                at com.google.devtools.build.lib.bazel.repository.downloader.ProgressInputStream.close(ProgressInputStream.java:129)
                at com.google.devtools.build.lib.bazel.repository.downloader.HashInputStream.close(HashInputStream.java:78)
                at java.base/java.io.SequenceInputStream.close(Unknown Source)
                at java.base/java.io.FilterInputStream.close(Unknown Source)
                at com.google.devtools.build.lib.bazel.repository.downloader.HttpDownloader.download(HttpDownloader.java:98)
```

### Which category does this issue belong to?

_No response_

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.


```
http_archive(
    name = "large-movie-review-dataset",
    build_file = "external/nlp/large-movie-review-dataset.BUILD",
    sha256 = "c40f74a18d3b61f90feba1e17730e0d38e8b97c05fde7008942e91923d1658fe",
    url = "file:///abc/nlp/sentiment/en/aclImdb_v1.tar.gz",
)
```

### Which operating system are you running Bazel on?


Linux

### What is the output of `bazel info release`?


release 9.1.0

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.


_No response_

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?


```text

```

### If this is a regression, please try to identify the Bazel commit where the bug was introduced with bazelisk --bisect.


This is a regression introduced in Bazel 9.

### Have you found anything relevant by searching the web?

_No response_

### Any other information, logs, or outputs that you want to share?

_No response_

--- a/src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/ProgressInputStream.java
+++ b/src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/ProgressInputStream.java
@@ -136,7 +136,7 @@ final class ProgressInputStream extends InputStream {
       return;
     }
     String via = "";
-    if (!url.getHost().equals(originalUrl.getHost())) {
+    if (url.getHost() != null && !url.getHost().equals(originalUrl.getHost())) {
       via = " via " + url.getHost();
     }
     eventHandler.post(new DownloadProgressEvent(originalUrl, url, bytesRead, totalBytes, false));

